controllers/operatorconfigmap_controller.go

```diff
@@ -591,3 +598,62 @@ func labelClientOperatorSubscription(c *OperatorConfigMapReconciler) error {
 	c.log.Info("successfully labelled ocs-client-operator subscription")
 	return nil
 }
+
+func (c *OperatorConfigMapReconciler) reconcileSubscription() error {
+
+	storageClients := &v1alpha1.StorageClientList{}
+	if err := c.list(storageClients); err != nil {
+		return fmt.Errorf("failed to list storageclients: %v", err)
+	}
+
+	var desiredChannel string
+	for idx := range storageClients.Items {
+		// empty if annotation doesn't exist or else gets desired channel
+		channel := storageClients.Items[idx].GetAnnotations()[utils.DesiredSubscriptionChannelAnnotationKey]
+		// skip clients with no/empty desired channel annotation
+		if channel != "" {
+			// check if we already established a desired channel
+			if desiredChannel == "" {
+				desiredChannel = channel
+			}
+			// check for agreement between clients
+			if channel != desiredChannel {
+				desiredChannel = ""
+				// two clients didn't agree for a same channel and no need to continue further
+				break
+			}
+		}
+	}
+	if desiredChannel == "" {
+		// no desired channel specified or no agreement between clients
+		return nil
```

Comment:
You are returning nil in all cases, and this is a non-error return value. In this type of cases, I would prefer to consolidate the return branching to a single return at the end of the function. You can achieve this easily by reverting the condition and nesting the happy path inside the if

Reply:
done.